compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -2885,8 +2885,35 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
     }
 
     private def genTryCatch(body: js.Tree, catches: List[CaseDef],
-        resultType: jstpe.Type,
-        isStat: Boolean)(implicit pos: Position): js.Tree = {
+        resultType: jstpe.Type, isStat: Boolean)(
+        implicit pos: Position): js.Tree = {
+
+      catches match {
+        case CaseDef(Ident(nme.WILDCARD), _, catchAllBody) :: Nil =>
+          genTryCatchCatchAll(body, catchAllBody, resultType, isStat)
+
+        case CaseDef(Typed(Ident(nme.WILDCARD), tpt), _, catchAllBody) :: Nil
+            if tpt.tpe.typeSymbol == ThrowableClass =>
+          genTryCatchCatchAll(body, catchAllBody, resultType, isStat)
+
+        case _ =>
+          genTryCatchNotCatchAll(body, catches, resultType, isStat)
+      }
+    }
+
+    private def genTryCatchCatchAll(body: js.Tree, catchAllBody: Tree,
```

Comment:
I'm a bit confused by this name: It's not only about catching everything, it's also about ignoring the result, right? Otherwise we couldn't take this shortcut.

Maybe call it something like `genTryCatchIgnoreAll`?

Reply:
Ah yes, that's a better name. I'll change it tomorrow.